src/pydvl/utils/types.py

```diff
@@ -95,3 +112,47 @@ def __call__(cls, *args, **kwargs):
 
     def create(cls, *args: Any, **kwargs: Any):
         return super().__call__(*args, **kwargs)
+
+
+Seed = Union[int, Generator]
+
+
+def ensure_seed_sequence(
+    seed: Optional[Union[Seed, SeedSequence]] = None
+) -> SeedSequence:
+    """
+    If the passed seed is a SeedSequence object then it is returned as is. If it is
+    a Generator the internal protected seed sequence from the generator gets extracted.
+    Otherwise, a new SeedSequence object is created from the passed (optional) seed.
+
+    Args:
+        seed: Either an int, a Generator object a SeedSequence object or None.
+
+    Returns:
+        A SeedSequence object.
+    """
+    if isinstance(seed, SeedSequence):
+        return seed
+    elif isinstance(seed, Generator):
+        return cast(SeedSequence, seed.bit_generator.seed_seq)  # type: ignore
+    else:
+        return SeedSequence(seed)
+
+
+def call_fn_multiple_seeds(
+    fn: Callable, *args, seeds: Tuple[Seed, ...], **kwargs
+) -> Tuple:
+    """
+    Execute a function multiple times with different seeds. It copies the arguments
+    and keyword arguments before passing them to the function.
+
+    Args:
+        fn: The function to execute.
+        args: The arguments to pass to the function.
+        seeds: The seeds to use.
+        kwargs: The keyword arguments to pass to the function.
+
+    Returns:
+        A tuple of the results of the function.
+    """
+    return tuple(fn(*deepcopy(args), **deepcopy(kwargs), seed=seed) for seed in seeds)
```

Comment:
I'm a bit wary of this hidden deepcopy. Datasets and Utilities are heavy objects. Would a user of this function expect the deep copy to happen?

Reply:
It is mostly for tests, moved to tests module.

Reply:
I wouldn't expect it to happen, but it would annoy me if it wasn't there :D